src/components/main/Account/Registration/index.tsx

```diff
@@ -2,53 +2,55 @@ import React, { useState } from 'react';
 import axios from 'axios';
 import { Link, useNavigate } from 'react-router-dom';
 import { FaUserCircle } from 'react-icons/fa';
+import * as accountService from "../../../../services/accountService";
 import "./index.css"
+import { useDispatch } from 'react-redux';
+import { setCurrentUser } from '../reducer';
 
 export default function Registration() {
-  const [fullName, setFullName] = useState('');
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
+  const [user, setUser] = useState<any>({});
   const [reenterPassword, setReenterPassword] = useState('');
-  const [error, setError] = useState('');
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
   const [isLoading, setIsLoading] = useState(false);
   const navigate = useNavigate();
```

Comment:
shouldn’t need dispatch or reducer when using mongoDB. maybe that could be part of the submission issue?

Reply:
Hmm I might be doing something wrong here then. I'm using dispatch to set the currentUser for the session I believe. I might be approaching that incorrectly.

Reply:
i’m probably wrong then. i’ll take a closer look tonight

Reply:
I sorta figured something out. For some reason, the first time I try to register for a given new user, it fails and the server crashes because the req doesn't have join_date or role even though I set it. So then the server crashes. But if I restart the service and try registering again it suddenly works...

Reply:
I fixed it in a sorta jancky manner, but it works